hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreScanner.java

```diff
@@ -130,6 +132,13 @@
    */
   public static final String HBASE_CELLS_SCANNED_PER_HEARTBEAT_CHECK =
       "hbase.cells.scanned.per.heartbeat.check";
+  /**
+   * The number of bytes after which we decide if to do only next() or continue
+   * with seek(). Currently it defaults to the STORESCANNER_PREAD_MAX_BYTES config
+   * To disable this feature put a value < 0.
+   */
+  public static final String HBASE_SWITCH_TO_NEXT_AFTER_BYTES_READ =
```

Comment:
The default HBASE_SWITCH_TO_NEXT_AFTER_BYTES_READ should smaller than one block size? But STORESCANNER_PREAD_MAX_BYTES is 4 times of block size.

Reply:
Currently am having 4 * blockSize as the default value now. I cannot have it less than one block size because unless I traverse multiple blocks I cannot make out this pattern. Within one block we cannot decide. I may be not getting your Q clearly - if that is the case do let me know I can try to explain it.

Reply:
I check your example in PPT. What this PR want to reduce is the compare times of "matcher.compareKeyForNextRow(nextIndexedKey, cell)", right?

Reply:
correct.

Reply:
The number of compares we do is the real reason for the perf regression. So we are trying to reduce that. When I say perf regression it is specifically for the case where the filter says SKIP but the SQM says NEXT_COL.  This will in general improve performance of a case where we have INLCUDE_NEXT_COL when we project all columns in a scan query. (that perf should be same in 1.x also).